contributors/design-proposals/nvml-and-volume-support.md

```diff
@@ -0,0 +1,84 @@
+# Proposal for NVML support and nvidia driver volume injection
+We propose leveraging the following to enhance the experimental GPU work:
+*  [NVML library](https://developer.nvidia.com/nvidia-management-library-nvml) to conduct GPU discovery
+* Nvidia driver volume injection
+
+## NVML support and dynamic loading
+Currently, GPU discovery is implemented by querying nvidia GPU devices under
+`/dev/nvidia*` directly, which may bring several problems. For example, it
+cannot detect GPU failures and  it only collects the entry of the device.  We
+think NVML is a more reliable and powerful method for GPU discovery, and advanced  GPU features (driver volume injection, GPU failure detection, heterogeneous GPU, topology-aware scheduling).
+
+We have found that dynamic loading is best when including the NVML library,
+because we can check for its existence at runtime. This allows a single kubelet
+binary to work on nodes that do not include NVML since we can skip all GPU logic.
+
+
+## Nvidia driver volume injection
+ Container-based applications  need Nvidia GPU drivers to interface with the
+ GPU. Installing the  GPU driver in the container is problematic  because the
+ GPU driver version must align with the host GPU driver version exactly for
+ proper functionality. Both host and container libraries need to be updated
+ simultaneously in this deployment.
+
+To simplify, we can borrow an idea from
```

Comment:
As described, I may prefer to see this defined more as a new type of `VolumeSource` that would be explicit in a pod specification rather than inferred based on a GPU request.

Are there approaches that we could use that do not require direct modification of the kubelet?  For example, could a DaemonSet be responsible for building this volume with drivers?  Personally, I would like to see us solve the GPU problem as a subset of the "i want to use an arbitrary device that requires local node topology awareness" rather than keep growing experimental more and more...